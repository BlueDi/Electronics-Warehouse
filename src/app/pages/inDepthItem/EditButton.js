import React, { Component } from 'react';
import { withCookies } from 'react-cookie';
import { Button } from 'semantic-ui-react';

/**
 * Creates buttons to edit an Item.
 * The interactions available depend on the user permissions.
 */
class EditButton extends Component {
  renderButton(cookie_name, button_text, onClick) {
    var canDo = this.props.cookies.get(cookie_name) === 'true';
    var positive = button_text === 'Accept' ? true : false;
    var negative = button_text === 'Cancel' ? true : false;
    return (
      canDo && (
        <Button
          key={button_text}
          onClick={onClick}
          positive={positive}
          negative={negative}
        >
          {button_text}
        </Button>
      )
    );
  }

  render() {
    return this.props.editing
      ? [
          this.renderButton('can_edit', 'Accept', this.props.handleAccept),
          this.renderButton('can_edit', 'Cancel', this.props.handleCancel)
        ]
      : [this.renderButton('can_edit', 'Edit', this.props.handleEdit)];
  }
}

export default withCookies(EditButton);
