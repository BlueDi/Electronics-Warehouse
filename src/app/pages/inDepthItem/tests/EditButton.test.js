import React from 'react';
import renderer from 'react-test-renderer';
import EditButton from '@pages/inDepthItem/EditButton';

describe('<EditButton />', () => {
  test('Renders properly', () => {
    const component = renderer.create(<EditButton />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders with edit enabled', () => {
    const component = renderer.create(<EditButton editing={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
