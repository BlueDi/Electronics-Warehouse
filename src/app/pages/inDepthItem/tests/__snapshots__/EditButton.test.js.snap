// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditButton /> Renders properly 1`] = `null`;

exports[`<EditButton /> Renders with edit enabled 1`] = `null`;
